src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
I have use cases for local file URL so I can't use the assertion on the general `Url` class, but it is workable and has been done for the app-specific URLs.
The one-param constructor `new Url("http://www.google.com/page")` is doable, but there is no use case for it as of now so I don't include it.